core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -4271,6 +4271,10 @@ public static int toInt(java.sql.Time v) {
     return v == null ? castNonNull(null) : toInt(v);
   }
 
+  // Method tagged as non-deterministic because it can throw.
+   // The DeterministicCodeOptimizer may otherwise try to lift it out of try-catch blocks.
+  // See https://issues.apache.org/jira/browse/CALCITE-6753
```

Comment:
Is this change should be added in CALCITE-6753?

Reply:
I don't have a fix for 6753, only a workaround